### PR TITLE
feat: 2.4.0

### PR DESCRIPTION
**Pull request by Pakisan**
- AsyncAPI 3.0.0
- Compatibility with IDEA 2023.3

### CODE REVIEW

File: build.gradle.kts
Comment (philCryoport) [inline build.gradle.kts:4]: v1.8.10 released Feb 2, 2023.

There's been [several Kotlin releases since](https://kotlinlang.org/docs/releases.html#release-details).  Why not a more recent one?

File: build.gradle.kts
Comment (philCryoport) [inline build.gradle.kts:57]: Might as well add `2023.3.2`, as it's listed [here](https://data.services.jetbrains.com/products?code=DG%2CPS%2CDL%2CMPS%2CMPSIIP%2CRDCPPP%2CWS%2CAC%2CIIC%2CIIE%2CGO%2CCL%2CPCD%2CPCC%2CRD%2CPCE%2CIIU%2CRM%2CPCP&fields=code%2Cname%2Creleases.downloads%2Creleases.version%2Creleases.build%2Creleases.type)

Expand node `15`
Expand `releases` sub-node, and `0` below that
...and you'll see `2023.3.2` as a listed entry

**Review by philCryoport — CHANGES_REQUESTED** [review]
Far too much to review with my limited time.  See comments above.

Next time you're going to do a wholesale object renaming, PLEASE...for love of everything is holy
...DON'T combine that wholesale object renaming with a AsyncAPI specification version update!?!
...AND DON'T combine that with a IntelliJ version roll!?!

File: build.gradle.kts
Comment (philCryoport) [inline build.gradle.kts:59]: Per https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#tasks-runpluginverifier-verifierversion this could be "LATEST"

File: build.gradle.kts
Comment (Pakisan) [reply build.gradle.kts:57]: It's early to add 2023.3.2, because is Release Candidate now

Will add this version after IDEA release

File: build.gradle.kts
Comment (Pakisan) [reply build.gradle.kts:4]: I don't use any kotlin-specific features in the plugin. So here is no need to bump version

File: build.gradle.kts
Comment (Pakisan) [reply build.gradle.kts:59]: Bad idea. Previously, after the update, build was broken because of bug. 

This is the main reason for me to keep a fixed version and after each update clean everything and try to build/run plugin from scratch

File: build.gradle.kts
Comment (philCryoport) [reply build.gradle.kts:4]: You might not, but the tests appear to use Kotlin...

I'll leave it to you to decide if this needs upgrading.

File: build.gradle.kts
Comment (philCryoport) [reply build.gradle.kts:59]: Ok

File: build.gradle.kts
Comment (philCryoport) [reply build.gradle.kts:57]: Ok

**Comment by philCryoport** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) **Quality Gate failed**
> 
> Failed conditions
> 
> [C Reliability Rating on New Code](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) (required ≥ A)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45)
> 
> idea Catch issues before they fail your Quality Gate with our IDE extension SonarLint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)

@Pakisan I suggest resolving this code-quality-gate. 

Then I'll approve the PR

**Comment by Pakisan** [comment]
> > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) **Quality Gate failed**
> > Failed conditions
> > [C Reliability Rating on New Code](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) (required ≥ A)
> > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45)
> > idea Catch issues before they fail your Quality Gate with our IDE extension SonarLint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)
> 
> @Pakisan I suggest resolving this code-quality-gate.
> 
> Then I'll approve the PR

Done

**Comment by Pakisan** [comment]
Will update kotlin version after IDEA 2023.3.2 release. 

Now I'm working on https://github.com/asyncapi/jasyncapi. Folks from Specmatic are waiting for new release with spec v3

**Comment by philCryoport** [comment]
> > > ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) **Quality Gate failed**
> > > Failed conditions
> > > [C Reliability Rating on New Code](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45) (required ≥ A)
> > > [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=asyncapi_jasyncapi-idea-plugin&pullRequest=45)
> > > idea Catch issues before they fail your Quality Gate with our IDE extension SonarLint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=pull-request)
> > 
> > 
> > @Pakisan I suggest resolving this code-quality-gate.
> > Then I'll approve the PR
> 
> Done

Thank you!  Just to be safe, I'm running the plugin verifier...

**Review by philCryoport — APPROVED** [review]
With note that Kotlin JVM version will be updated in a separate PR.

Successful clean / buildPlugin / runPluginVerifier.

**Review by theschles — APPROVED** [review]
This is same person as @philCryoport -- that's my current work account, this is my personal account.

**Pull request merged**
